nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/PublishJMS.java

```diff
@@ -131,4 +143,10 @@ protected JMSPublisher finishBuildingJmsWorker(CachingConnectionFactory connecti
         session.read(flowFile, in -> StreamUtils.fillBuffer(in, messageContent, true));
         return messageContent;
     }
+
+    private String extractTextMessageBody(FlowFile flowFile, ProcessSession session) {
+        final StringWriter writer = new StringWriter();
+        session.read(flowFile, in -> IOUtils.copy(in, writer, Charset.defaultCharset()));
```

Comment:
We probably want to expose the character set as a property. At a minimum, we should be explicit in defining which character set would be used (UTF-8 perhaps) instead of relying on the default charset, as this would produce different results from environment to environment.

Reply:
I did think about this, and I chose this approach on purpose.  It's kind of the whole point of setting a default Charset in your JVM, to produce results that are specific to your desired environment.  Is this not right?

Reply:
i'm not sure if this is a helpful "middle ground" suggestion or not, but you could add a charset property that itself defaults to Charset.defaultCharset().  That way the out of the box behavior uses the charset specified for the jvm, but can be overridden on a per processor basis if needed.

Reply:
Oh, I didn't think about that approach!  I like it.  I'll make the changes.

Reply:
And now that I look closer, I see there is already a precedent for this in the code, such as in LogAttribute.